packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -217,12 +211,12 @@ class OAuthController {
             throw e;
         }
 
-        const session: OAuthSession = this.sessionStore[state as string] as OAuthSession;
+        const session = (await cache.get(state as string)) as OAuthSession;
 
         if (session == null) {
             throw new Error('No session found for state: ' + state);
         } else {
-            delete this.sessionStore[state as string];
+            await cache.delete(state as string);
```

Comment:
It's good that we are deleting the session here.

We also need to take care of expiring the sessions that do not complete so that the # of DB records doesn't grow. Maybe that could be done on startup, e.g. we erase the sessions that are more than 24h old?